### PR TITLE
Ctb 10 update an issue

### PR DESCRIPTION
**Pull request by ghost**
sloppy way of handling a PUT issue, but should work regardless

### CODE REVIEW

File: routers/issueRouter.js [hunk 1]
```diff
@@ -42,4 +42,16 @@ router.post( '/', jsonParser, function( req, res ){
 	});
 });
 
-module.exports = router;
+// PUT issue, updating a current issue
+router.put( '/', jsonParser, function( req, res){
+	var issue = req.body;
+	db.findByIdAndUpdate(issue._id, { name: issue.name, description: issue.description, location: { lat: issue.location[lat], long: issue.location[long]}, priority: issue.priority, open: issue.open, reportedDate: issue.reportedDate, resolvedDate: issue.resolvedDate}, function( err, doc ){
```
Comment (anthonynatoli) [inline routers/issueRouter.js]: I think findByIdandUpdate only needs the fields that will be updated. So could try something like:
`var id = issue._id`
`delete issue._id; //this removes the _id field from the issue object`
`db.findByIdAndUpdate( id, issue, function( ... ){ ... } );`
In the future. That way the client only passes in attributes of the issue that changed.

**Comment by anthonynatoli** [comment]
@tomaskratka If it works, go ahead and merge it! Keep that change I recommended in mind. I may try to make that change today before the meeting.

File: routers/issueRouter.js (hunk 1 above)
Comment (ghost) [inline routers/issueRouter.js]: we can just pass in the schema object? I was under the assumption that you had to construct that in the findbyidandupdate call. I can make that change

Comment (anthonynatoli) [inline routers/issueRouter.js]: The clients should be abiding by the structure of the schema object anyway, so I'm imagining the put request body to be something like:
    `{ _id: some_id, attribute_to_change: new_value, another_attribute_to_change: another_new_value, ...}`

I don't think it needs the entire object though, just the changed keys. At least that's how I'm understanding it from their documentation (that isn't all that detailed). I'll test it extensively later to make sure it works.

**Pull request merged**